sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -66,8 +65,7 @@ private[spark] class HiveExternalCatalog(client: HiveClient) extends ExternalCat
     try {
       body
     } catch {
-      case e: NoSuchItemException =>
-        throw new AnalysisException(e.getMessage)
+      case e: AnalysisException => throw new AnalysisException(e.getMessage)
```

Comment:
It would have had to to throw `AnalysisException`. I suspect there's no reason now to lose the original stack trace.